conn.go

```diff
@@ -152,10 +161,18 @@ func Connect(conn io.ReadWriteCloser, opts ...func(*Conn) error) (*Conn, error)
 		return nil, err
 	}
 
+	connection, isNetConn := conn.(net.Conn)
+	deadline, ok := ctx.Deadline()
+	if ok && isNetConn {
+		connection.SetReadDeadline(deadline)
+	}
 	response, err := reader.Read()
 	if err != nil {
 		return nil, err
 	}
+	if ok && isNetConn {
```

Comment:
isn't this the same as the check on line 165? I think you meant to use `!ok`
also why `SetReadDeadline` and not `SetDeadline`?

Reply:
additionaly, can we move this one above so it's next to the other if

Reply:
The problem to be solved here is that after the stomp client sends the stomp protocol connect message, the stomp server does not reply due to some reasons, such as: stomp server resources are insufficient stomp client message, and the stomp client will block forever. The use of SetReadDeadline here is correct because we want reader.Read() to time out instead of blocking forever, Calling SetReadDeadline again at line 173 with the argument time.Time{} restores net.Conn's read blocking behavior because we don't want it to affect subsequent data interactions. thank you

Reply:
Similar writing can be found in the standard library http, src/net/http/server.go can find many similar implementation methods, such as go1.23.1 （src/net/http/server.go）:
![20241024-094716](https://github.com/user-attachments/assets/d59c9fc5-5174-47cc-b89e-4cc4675a812c)